tests/unit/utils/test_logger.py

```diff
@@ -45,8 +49,27 @@ def test_logger_show_message_flag(mock_load_dotenv, capsys):
 
     # check captured log output
     # the log message should not be captured due to log level
-    # but the message should be shown
     captured_out = capsys.readouterr().out
     assert "Set LOG_LEVEL environment variable (e.g., INFO, DEBUG)" in captured_out
-    captured_err = capsys.readouterr().out
+    assert "Debug message" not in captured_out
+
+    # error level should be empty
+    captured_err = capsys.readouterr().err
+    assert captured_err == ""
+
+
+@patch("dotenv.load_dotenv")
```

Comment:
why isn't this `ols.utils.logger.dotenv.load_dotenv`?

or alternatively why isn't this:
```
@patch("ols.app.endpoints.ols.LLMChain", new=mock_llm_chain("test response"))
```
(from https://github.com/openshift/lightspeed-service/blob/3bc9ded2b6b5ce60283b4235cf39fd40730ffa10/tests/integration/test_ols.py#L51 )

```
@patch("ols.src.llms.llm_loader.LLMLoader", new=mock_llm_chain("test response"))
```

instead?

In the dotenv case you seem to be patching the lib directly.  In the case i pointed to you seem to be patching the variable value at the point of use.